Ladybird/LocationEdit.cpp

```diff
@@ -27,6 +28,7 @@ void LocationEdit::focusInEvent(QFocusEvent* event)
 {
     QLineEdit::focusInEvent(event);
     highlight_location();
+    QTimer::singleShot(0, this, &QLineEdit::selectAll);
```

Comment:
Hmm, why is this on a single-shot timer instead of immediate?

Reply:
iʼd guess it waits for event sent at the end of `highlight_location` to get propagated, i.e. pushing an event onto event loop after it (so the current text would become a _commit string_?). yet iʼm also not sure if that is needed, if you were interested in that.

thereʼs a possibility to explore here: would the `Selection` attribute on that event do the work? (properly set—got problems reading english docs there! 😣)

Reply:
According to [this](https://stackoverflow.com/a/51807268) calling selectAll directly won't work.